### PR TITLE
Allow user tags on virtual machine

### PR DESCRIPTION
**Pull request by reachjainrahul**
## Description
In addition to discovered tags from cloud, we now allow user tags on virtual machine objects. These user tags will also be translated to external entity labels, which can be used for Antrea NetworkPolicy configuration.

## Changes
New update handler is added in virtual machine rest handler to update the virtual machine in-memory cache. In case of restart, we use external entity object of the virtual machine to re-generate user tags.

### CODE REVIEW

File: pkg/apiserver/registry/inventory/virtualmachine/rest.go [hunk 1]
```diff
@@ -83,6 +84,37 @@ func (r *REST) ShortNames() []string {
 	return []string{"vm"}
 }
 
+func (r *REST) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, _ rest.ValidateObjectFunc,
```
Comment (Anandkumar26) [inline pkg/apiserver/registry/inventory/virtualmachine/rest.go:87]: nit: Add a comment saying only spec fields can be updated via the rest interface.

File: pkg/inventory/inventory.go [hunk 2]
```diff
@@ -323,3 +323,9 @@ func (i *Inventory) compareVirtualMachineObjects(cached, discovered runtimev1alp
 	sortInterfacesFunc(dis.NetworkInterfaces)
 	return reflect.DeepEqual(cached.NetworkInterfaces, dis.NetworkInterfaces)
 }
+
+// UpdateVm updates virtual machine object in vm cache.
+func (i Inventory) UpdateVm(vm *runtimev1alpha1.VirtualMachine) error {
```
Comment (Anandkumar26) [inline pkg/inventory/inventory.go:328]: nit: func (i *Inventory)

**Review by Anandkumar26 — COMMENTED** [review]
Minor comment, once addressed we can merge it.

File: pkg/apiserver/registry/inventory/virtualmachine/rest.go (hunk 1 above)
Comment (reachjainrahul) [reply pkg/apiserver/registry/inventory/virtualmachine/rest.go:87]: its kind of expected.. Status is runtime, and Spec is desired.. So not required.

**Review by shenmo3 — APPROVED** [review]

**Pull request merged**